### PR TITLE
feat: add reviewer-test-quality agent for automated test quality evaluation

### PR DESCRIPTION
**Pull request by TheTarry**
Closes #187

## Summary
- Added `reviewer-test-quality` subagent at `agents/reviewer-test-quality.md` — evaluates test quality across spec alignment, edge case coverage, DAMP principles, and testing anti-patterns
- Integrated into `reviewing-code-systematically` skill with conditional dispatch (only dispatched when test files exist in scope)
- Recorded architecture decision in `docs/decisions/2026-07-reviewer-test-quality-agent.md`

## Test Plan
- [x] Frontmatter matches existing reviewer agent patterns (verified against reviewer-dead-code)
- [x] All 4 reviewer references present in reviewing-code-systematically skill
- [x] Agent name consistent between agent file and skill dispatch
- [x] Code review passed (reviewer-dead-code: PASS, reviewer-docs: SKIP, reviewer-newcomer: NEEDS_CHANGES → fixed)

### CODE REVIEW

File: agents/reviewer-test-quality.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+---
+name: reviewer-test-quality
+description: Use when reviewing test files for quality — spec alignment, edge case coverage, DAMP principles, and testing anti-pattern detection (excessive mocking, testing implementation details, etc.).
+mode: subagent
+permission:
+  edit: deny
+  bash:
+    "git diff*": allow
+    "npm test*": allow
+    "yarn test*": allow
```
Comment (TheTarry) [inline agents/reviewer-test-quality.md]: Why these bash script permissions specifically? Neither `npm` nor `yarn` are part of my standard build tooling.

It was my understanding this new agent would review tests, rather than execute them?
